WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -136,13 +137,48 @@ import Foundation
         refreshTableViewHeaderLayout()
     }
 
+    public override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "handleApplicationDidBecomeActive:", name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
 
-    // MARK: -
+    public override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
 
     public override func preferredStatusBarStyle() -> UIStatusBarStyle {
         return .LightContent
     }
 
+    public override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+
+        if needsRefreshCachedCellHeightsBeforeLayout {
+            needsRefreshCachedCellHeightsBeforeLayout = false
+
+            let width = view.frame.width
+            tableViewHandler.refreshCachedRowHeightsForWidth(width)
+            tableView.reloadRowsAtIndexPaths(tableView.indexPathsForVisibleRows!, withRowAnimation: .None)
+        }
+    }
+
+    @available(iOS 8.0, *)
```

Comment:
Wooow this one is new to me. Is that what i think it is? an annotation marking the method "visible only" to iOS 8?